net/dns/manager.go

```diff
@@ -137,6 +139,47 @@ func (m *Manager) Set(cfg Config) error {
 	return nil
 }
 
+// compileHostEntries creates a list of single-label resolutions possible
+// from the configured hosts and search domains.
+// The entries are compiled in the order of the search domains, then the hosts.
+// The returned list is sorted by the first hostname in each entry.
+func compileHostEntries(cfg Config) (hosts []*HostEntry) {
+	didLabel := make(map[string]bool, len(cfg.Hosts))
+	for _, sd := range cfg.SearchDomains {
+		for h, ips := range cfg.Hosts {
+			if !sd.Contains(h) || h.NumLabels() != (sd.NumLabels()+1) {
+				continue
+			}
+			ipHosts := []string{string(h.WithTrailingDot()), ""}[:1]
+			if label := dnsname.FirstLabel(string(h)); !didLabel[label] {
+				mak.Set(&didLabel, label, true)
```

Comment:
didLabel was initialized explicitly, no need to use magical mak.Set here, `didLabel[label] = true` works